### PR TITLE
gh-93649: Split tracemalloc tests from _testcapimodule.c

### PR DESCRIPTION
**Pull request by erlend-aasland**
Also split memory tests from Lib/test/test_capi/test_misc.py

* Issue: gh-93649

### CODE REVIEW

**Review by sobolevn — APPROVED** [review]
I will have so many merge conflicts in my C coverage PRs, ahaha 🙈 

Thanks for working on this!

**Review by vstinner — APPROVED** [review]
LGTM.

**Review by ezio-melotti — COMMENTED** [review]
The splits don't seem to preserve history.
For `test_mem.py` it should be possible to preserve it since it's a new file.  For `mem.c`, I think `git` is supposed to figure it out on its own since the code have been moved but no files have been created.  Maybe moving the code in a commit on its own will work?

**Comment by erlend-aasland** [comment]
> The splits don't seem to preserve history. For `test_mem.py` it should be possible to preserve it since it's a new file. For `mem.c`, I think `git` is supposed to figure it out on its own since the code have been moved but no files have been created. Maybe moving the code in a commit on its own will work?

Tests with various `git log` commands:

```console
$ git log --oneline --follow -S hook_malloc Modules | wc -l
4
```

OTOH, `git log -L` seems to have problems, but that command does not work with `--follow`:

```console
$ git log --oneline -L:hook_malloc:Modules/_testcapi/mem.c | wc -l
1
```

> Maybe moving the code in a commit on its own will work?

Possibly, but I won't bet money on it since the commits are squashed upon merge anyway; previous move operations in `Lib/test` have worked swell (and there's been a lot of that; Serhiy was busy with test organization fairly recently). IMO reorganizing the code like this has far more benefits than disadvantages, since the resulting smaller files are easier to read, understand, and maintain, which again makes it easier to fix bugs.

File: Lib/test/test_capi/test_mem.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+@unittest.skipUnless(support.Py_DEBUG, 'need Py_DEBUG')
+class PyMemDefaultTests(PyMemDebugTests):
+    # test default allocator of Python compiled in debug mode
+    PYTHONMALLOC = ''
+
+
+if __name__ == "__main__":
+    unittest.main()
+
```
Comment (vstinner) [inline Lib/test/test_capi/test_mem.py]: You may remove the empty line at the end :-)

**Review by vstinner — APPROVED** [review]
LGTM.

File: Lib/test/test_capi/test_mem.py (hunk 1 above)
Comment (erlend-aasland) [reply Lib/test/test_capi/test_mem.py]: Oops 😆

**Review by kumaraditya303 — APPROVED** [review]
LGTM

**Pull request merged**